be2-scala/src/test/scala/ch/epfl/pop/decentralized/MonitorSuite.scala

```diff
@@ -0,0 +1,80 @@
+package ch.epfl.pop.decentralized
+
+import akka.actor.ActorSystem
+import akka.stream.scaladsl.Source
+import akka.testkit.{TestKit, TestProbe}
+import ch.epfl.pop.model.network.method.ParamsWithMap
+import ch.epfl.pop.model.network.{JsonRpcRequest, MethodType}
+import ch.epfl.pop.pubsub.graph.validators.RpcValidator
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.funsuite.{AnyFunSuiteLike => FunSuiteLike}
+import org.scalatest.matchers.should.Matchers
+import util.examples.JsonRpcRequestExample
+import scala.concurrent.duration.DurationInt
+
+class MonitorSuite extends TestKit(ActorSystem("MonitorSuiteActorSystem")) with FunSuiteLike with Matchers with BeforeAndAfterAll {
+
+  override def afterAll(): Unit = {
+    // Stops the test actor system
+    TestKit.shutdownActorSystem(system)
+  }
+
+  test("monitor should start scheduling heartbeats when told a server connected") {
+
+    val testProbe = TestProbe()
+    val monitorRef = system.actorOf(
+      Monitor.props(testProbe.ref, heartbeatRate = 2.seconds, messageDelay = 1.seconds)
```

Comment:
These test settings variables (n seconds) are repeated across the tests so they could gain consistency by becoming attributes of the class